rules/alerting.go

```diff
@@ -538,3 +538,53 @@ func (r *AlertingRule) HTMLSnippet(pathPrefix string) html_template.HTML {
 func (r *AlertingRule) HoldDuration() time.Duration {
 	return r.holdDuration
 }
+
+// Equals return if two alerting rule has the same
+// 1. name
+// 2. expr
+// 3. duration
+// 4. labels
+// 5. annotations
+// 6. externalLabels
+func (r *AlertingRule) Equals(rule Rule) bool {
+	ar, ok := rule.(*AlertingRule)
+	if !ok {
+		return false
+	}
+
+	if r.name != ar.name {
+		return false
+	}
+
+	if r.vector.String() != ar.vector.String() {
+		return false
+	}
+
+	if r.holdDuration != ar.holdDuration {
+		return false
+	}
+
+	if !labels.Equal(r.labels, ar.labels) {
+		return false
+	}
+
+	if !labels.Equal(r.annotations, ar.annotations) {
+		return false
+	}
+
+	if len(r.externalLabels) != len(ar.externalLabels) {
+		return false
+	}
+
+	for k, v := range r.externalLabels {
```

Comment:
if !labels.Equal(r.externalLabels, ar.externalLabels) {

would that work??

Reply:
It is a `map[string]string` type

Reply:
yeah. it wont work